src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -262,18 +309,20 @@ private void UnregisterRuleSetEventHandler()
 
         private void TrackBulkFileOperations()
         {
+            RoslynDebug.AssertNotNull(_workspace);
+
             // we will pause whatever ambient work loads we have that are tied to IGlobalOperationNotificationService
             // such as solution crawler, pre-emptive remote host synchronization and etc. any background work users didn't
             // explicitly asked for.
             //
             // this should give all resources to BulkFileOperation. we do same for things like build, 
             // debugging, wait dialog and etc. BulkFileOperation is used for things like git branch switching and etc.
-            var globalNotificationService = _workspace.Services.GetService<IGlobalOperationNotificationService>();
+            IGlobalOperationNotificationService globalNotificationService = _workspace.Services.GetRequiredService<IGlobalOperationNotificationService>();
 
             // BulkFileOperation can't have nested events. there will be ever only 1 events (Begin/End)
             // so we only need simple tracking.
-            var gate = new object();
-            GlobalOperationRegistration localRegistration = null;
+            object gate = new object();
```

Comment:
why?

Reply:
The value is used in a local function.

Reply:
➡️ Covered by discussion in https://github.com/dotnet/roslyn/pull/33323#discussion_r398316375